docs/development/core/server/kibana-plugin-server.uisettingsparams.md

```diff
@@ -1,28 +1,29 @@
-<!-- Do not edit this file. It is automatically generated by API Documenter. -->
-
-[Home](./index.md) &gt; [kibana-plugin-server](./kibana-plugin-server.md) &gt; [UiSettingsParams](./kibana-plugin-server.uisettingsparams.md)
-
-## UiSettingsParams interface
-
-UiSettings parameters defined by the plugins.
-
-<b>Signature:</b>
-
-```typescript
-export interface UiSettingsParams 
-```
-
-## Properties
-
-|  Property | Type | Description |
-|  --- | --- | --- |
-|  [category](./kibana-plugin-server.uisettingsparams.category.md) | <code>string[]</code> | used to group the configured setting in the UI |
-|  [description](./kibana-plugin-server.uisettingsparams.description.md) | <code>string</code> | description provided to a user in UI |
-|  [name](./kibana-plugin-server.uisettingsparams.name.md) | <code>string</code> | title in the UI |
-|  [optionLabels](./kibana-plugin-server.uisettingsparams.optionlabels.md) | <code>Record&lt;string, string&gt;</code> | text labels for 'select' type UI element |
-|  [options](./kibana-plugin-server.uisettingsparams.options.md) | <code>string[]</code> | array of permitted values for this setting |
-|  [readonly](./kibana-plugin-server.uisettingsparams.readonly.md) | <code>boolean</code> | a flag indicating that value cannot be changed |
-|  [requiresPageReload](./kibana-plugin-server.uisettingsparams.requirespagereload.md) | <code>boolean</code> | a flag indicating whether new value applying requires page reloading |
-|  [type](./kibana-plugin-server.uisettingsparams.type.md) | <code>UiSettingsType</code> | defines a type of UI element [UiSettingsType](./kibana-plugin-server.uisettingstype.md) |
-|  [value](./kibana-plugin-server.uisettingsparams.value.md) | <code>SavedObjectAttribute</code> | default value to fall back to if a user doesn't provide any |
-
+<!-- Do not edit this file. It is automatically generated by API Documenter. -->
+
+[Home](./index.md) &gt; [kibana-plugin-server](./kibana-plugin-server.md) &gt; [UiSettingsParams](./kibana-plugin-server.uisettingsparams.md)
+
+## UiSettingsParams interface
+
+UiSettings parameters defined by the plugins.
```

Comment:
The generated doc files are using windows line feed (this is how the tool work so we decided to keep it like this), and you pushed them with LF (autocrlf probably). Please regenerate the doc and commit them with preserved line feeds.

Reply:
I don't think there is a way to configure it currently https://github.com/elastic/kibana/issues/53761#issuecomment-573644536

Reply:
I don't think either, that's why we need to stick to CRLF for now. Issue is, a lot of developers uses `core.autocrlf input` git config to convert to LF (which is a good practice most of the time) and may push without even knowing they altered the linefeeds.

Reply:
we can enforce EOL format for all `*.md` files on the github repo level https://github.com/microsoft/rushstack/blob/2c941e21777393c67979b25dc087433485b12575/.gitattributes#L44

Reply:
Obviously I know very little about this tool, but I'd be tempted to wrap it in a script that converts the line endings. Having markdown files that _must_ use different line endings is going to cause some headaches.